### PR TITLE
Add fastapi integration testing tutorial

### PR DESCRIPTION
**Pull request by DarkStarStrix**
# Added FastAPI Integration Testing Tutorial

This PR adds a comprehensive tutorial for writing integration tests with FastAPI, including:
- Database setup with SQLAlchemy
- Authentication testing
- Protected routes testing
- Best practices

## Checklist
- [x] Included documentation when adding new features.
- [x] Included new tests or update existing tests when applicable. 
- [x] Allow maintainers to push and squash when merging my commits.

```
Added tutorial documentation for FastAPI integration testing.
```

Added myself (Allan)

### CODE REVIEW

**Comment by The-Compiler** [comment]
The pytest documentation seems like the wrong place for this - FastAPI solves a very specific problem, which will only have a small overlap with what people use pytest for.

Maybe the FastAPI docs would be a better place, or some sort of blog post?

File: doc/en/how-to/fastapi-integration-testing-tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+This tutorial demonstrates how to write integration tests for FastAPI applications using pytest.
+
+Prerequisites
+------------
+
+* Python 3.7+
+* pytest
+* FastAPI
+* SQLAlchemy
+* pytest-asyncio
```
Comment (graingert) [inline doc/en/how-to/fastapi-integration-testing-tutorial.rst:13]: FastAPI already comes with a pytest plug-in, just use `@pytest.mark.anyio` you don't need to install an extra one

File: doc/en/how-to/fastapi-integration-testing-tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        )
+        assert response.status_code == 401
+
+5. **Async Support**: For async endpoints, use ``pytest-asyncio``:
+
+.. code-block:: python
+
+    @pytest.mark.asyncio
+    async def test_async_endpoint(client):
+        response = await client.get("/async-endpoint")
```
Comment (graingert) [inline doc/en/how-to/fastapi-integration-testing-tutorial.rst:181]: This client.get isn't async

File: doc/en/how-to/fastapi-integration-testing-tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+In ``conftest.py``, create fixtures for your test database:
+
+.. code-block:: python
+
+    import pytest
+    from sqlalchemy import create_engine
+    from sqlalchemy.orm import sessionmaker
+    from fastapi.testclient import TestClient
+    from app.main import app
```
Comment (graingert) [inline doc/en/how-to/fastapi-integration-testing-tutorial.rst:40]: When testing it's best to use the create_app pattern so you create a new instance of app for each test

**Comment by graingert** [comment]
The FastAPI docs is the best place for this and it's already covered with https://fastapi.tiangolo.com/tutorial/testing/

Perhaps you could ask if there's space for documentation on using httpx.AsyncClient, ASGITransport, asgi-lifespan and the anyio plugin in the FastAPI discussion forums

**Comment by DarkStarStrix** [comment]
Hey guys, thanks for the feedback. Let me explain my reasoning for this 1. I was looking around the pytest docs and did not find a tutorial on integration tests and thought it would be a great contribution. I think you are right with the fast-api part and there are already documents written I might do a blog post instead

**Comment by webknjaz** [comment]
@DarkStarStrix it might be useful to point to various plugin and framework docs. And have a generic integration testing example. But it shouldn't be framework-specific because it assumes knowledge of that framework and puts additional burden on the maintainers. If one framework guide is present, why not have another 10 framework guides, then?

**Pull request closed without merging**